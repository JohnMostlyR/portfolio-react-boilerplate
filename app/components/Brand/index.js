/**
 * Brand Component
 * @param {string} [href='/']
 * @param {string} [showNameBreakpoint='375px']
 * @returns {*}
 * @constructor
 */
import React from 'react';
import PropTypes from 'prop-types';
import { FormattedMessage } from 'react-intl';

import messages from './messages';
import Wrapper from './Wrapper';
import BrandLink from './BrandLink';
import Logo from '../../components/Logo';

const Brand = ({ href, showNameBreakpoint }) => (
  <Wrapper>
    <FormattedMessage {...messages.title}>
      {message => (
        <BrandLink
          href={href}
          showNameBreakpoint={showNameBreakpoint}
          title={message}
        >
          <Logo showNameBreakpoint={showNameBreakpoint} />
        </BrandLink>
      )}
    </FormattedMessage>
  </Wrapper>
);

Brand.propTypes = {
  href: PropTypes.string,
  showNameBreakpoint: PropTypes.string,
};

Brand.defaultProps = {
  href: '/',
  showNameBreakpoint: '375px',
};

export default Brand;
