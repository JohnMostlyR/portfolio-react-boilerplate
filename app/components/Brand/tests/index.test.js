import React from 'react';
import { shallow } from 'enzyme';

import Brand from '../index';

describe('<Brand />', () => {
  it('should render and match the snapshot', () => {
    const wrapper = shallow(<Brand />);
    expect(wrapper).toMatchSnapshot();
  });
});
