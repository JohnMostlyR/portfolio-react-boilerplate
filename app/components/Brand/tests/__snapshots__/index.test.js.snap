// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Brand /> should render and match the snapshot 1`] = `
<Wrapper>
  <FormattedMessage
    defaultMessage="← Back to the homepage"
    id="portfolio.component.BrandLink.title"
    values={Object {}}
  />
</Wrapper>
`;
